stellargraph/layer/deep_graph_infomax.py

```diff
@@ -114,7 +114,8 @@ def build(self):
 
         x_inp, node_feats = self.base_model.build(multiplicity=1)
         # identity layer so we can attach a name to the tensor
-        node_feats = Lambda(lambda x: x, name=self._unique_id)(node_feats)
+        rand_string = "".join(random.choice(ascii_lowercase) for _ in range(5))
```

Comment:
What's this for?

Reply:
I noticed that `keras` gives every layer a unique name:

```
import tensorflow as tf

print(tf.keras.layers.Input(5).name, tf.keras.layers.Input(5).name)
```

 I think this is so that if you combine layers from different models no two layers will have the same name - which would throw an error. 

This is to try and mimic that behaviour, but at the moment its kinda pointless because of the mutation behaviour in GCN that I talk about below.